src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -490,27 +490,72 @@ SendCommandListToWorkerOutsideTransaction(const char *nodeName, int32 nodePort,
  * coordinated transaction. Any failures aborts the coordinated transaction.
  */
 void
-SendMetadataCommandListToWorkerInCoordinatedTransaction(const char *nodeName,
-														int32 nodePort,
-														const char *nodeUser,
-														List *commandList)
+SendMetadataCommandListToWorkerListInCoordinatedTransaction(List *workerNodeList,
```

Comment:
@marcocitus do you see any drawbacks of this function? I'm mostly asking regarding failure handling. All seems good to me, but I'd appreciate if you can also have a quick look